sdk/tools/cabman/cabinet.cxx

```diff
@@ -591,54 +562,49 @@ ULONG CCabinet::FindNext(PCAB_SEARCH Search)
  */
 {
     bool bFound = false;
-    PSEARCH_CRITERIA Criteria;
     ULONG Status;
 
     if (RestartSearch)
     {
-        Search->Next  = FileListHead;
+        Search->Next  = FileList.begin();
 
         /* Skip split files already extracted */
-        while ((Search->Next) &&
-            (Search->Next->File.FileControlID > CAB_FILE_MAX_FOLDER) &&
-            (Search->Next->File.FileOffset <= LastFileOffset))
+        while ((Search->Next != FileList.end()) &&
+            ((*Search->Next)->File.FileControlID > CAB_FILE_MAX_FOLDER) &&
+            ((*Search->Next)->File.FileOffset <= LastFileOffset))
         {
             DPRINT(MAX_TRACE, ("Skipping file (%s)  FileOffset (0x%X)  LastFileOffset (0x%X).\n",
-                Search->Next->FileName.c_str(), (UINT)Search->Next->File.FileOffset, (UINT)LastFileOffset));
-            Search->Next = Search->Next->Next;
+                (*Search->Next)->FileName.c_str(), (UINT)(*Search->Next)->File.FileOffset, (UINT)LastFileOffset));
+            Search->Next++;
         }
 
         RestartSearch = false;
     }
 
     /* Check each search criteria against each file */
-    while(Search->Next)
+    while(Search->Next != FileList.end())
```

Comment:
```suggestion
    while (Search->Next != FileList.end())
```

Reply:
I just updated the code to work with lists, I did not re-format the entire module so I'd prefer to keep formatting changes out of this.